aiohttp/web_response.py

```diff
@@ -323,7 +330,12 @@ async def _do_start_compression(self, coding: ContentCoding) -> None:
         if coding != ContentCoding.identity:
             assert self._payload_writer is not None
             self._headers[hdrs.CONTENT_ENCODING] = coding.value
-            self._payload_writer.enable_compression(coding.value)
+            if isinstance(self._payload_writer, StreamWriter):
+                self._payload_writer.enable_compression(
+                    coding.value, self._compression_strategy
+                )
+            else:
+                self._payload_writer.enable_compression(coding.value)
```

Comment:
Tests are passing now. I'd still like @asvetlov or someone to confirm if we should use the `isinstance()` check (or change the type to `StreamWriter` or add the parameter to `AbstractStreamWriter`) before merging.

Reply:
From what I can see, the abstract class is not used in our code. It is only a StreamWriter within our internal code, so I don't even see a way for anyone to replace it. So, I suspect the abstract is not really needed. For now, let's just change the abstract class.

Reply:
```suggestion
            self._payload_writer.enable_compression(
                coding.value, self._compression_strategy
            )
```